Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Unix.swift

```diff
@@ -0,0 +1,286 @@
+            let rawByteValue: UnsafeMutablePointer<CChar> = valueContainer.createRawBytes()
+            defer {
+                rawByteKey.deallocate()
+                rawByteValue.deallocate()
+            }
+            /// length = `key` + `=` + `value` + `\null`
+            let totalLength = keyContainer.count + 1 + valueContainer.count + 1
+            let fullString: UnsafeMutablePointer<CChar> = .allocate(capacity: totalLength)
+            #if canImport(Darwin)
+            _ = snprintf(ptr: fullString, totalLength, "%s=%s", rawByteKey, rawByteValue)
+            #else
+            _ = _shims_snprintf(fullString, CInt(totalLength), "%s=%s", rawByteKey, rawByteValue)
+            #endif
+            return fullString
+        }
+
+        var env: [UnsafeMutablePointer<CChar>?] = []
+        switch self.config {
+        case .inherit(let updates):
+            var current = ProcessInfo.processInfo.environment
+            for (keyContainer, valueContainer) in updates {
+                if let stringKey = keyContainer.stringValue {
+                    // Remove the value from current to override it
+                    current.removeValue(forKey: stringKey)
+                }
+                // Fast path
+                if case .string(let stringKey) = keyContainer,
+                   case .string(let stringValue) = valueContainer {
+                    let fullString = "\(stringKey)=\(stringValue)"
+                    env.append(strdup(fullString))
+                    continue
+                }
+
+                env.append(createFullCString(fromKey: keyContainer, value: valueContainer))
+            }
+            // Add the rest of `current` to env
+            for (key, value) in current {
+                let fullString = "\(key)=\(value)"
+                env.append(strdup(fullString))
+            }
+        case .custom(let customValues):
+            for (keyContainer, valueContainer) in customValues {
+                // Fast path
+                if case .string(let stringKey) = keyContainer,
+                   case .string(let stringValue) = valueContainer {
+                    let fullString = "\(stringKey)=\(stringValue)"
+                    env.append(strdup(fullString))
+                    continue
+                }
+                env.append(createFullCString(fromKey: keyContainer, value: valueContainer))
+            }
+        }
+        env.append(nil)
+        return env
+    }
+}
+
+// MARK: Args Creation
+extension Subprocess.Arguments {
+    // This method follows the standard "create" rule: `args` needs to be
+    // manually deallocated
+    internal func createArgs(withExecutablePath executablePath: String) -> [UnsafeMutablePointer<CChar>?] {
+        var argv: [UnsafeMutablePointer<CChar>?] = self.storage.map { $0.createRawBytes() }
+        // argv[0] = executable path
+        if let override = self.executablePathOverride {
```

Comment:
Same goes here - `Task` previously used the `fileSystemRepresentation` for all string arguments rather than just using the as-is cString value. Do we want to do the same here or diverge for `Subprocess`?